scs2-definition/src/main/java/us/ihmc/scs2/definition/robot/urdf/URDFTools.java

```diff
@@ -209,66 +247,256 @@ public static URDFModel loadURDFModel(InputStream inputStream, Collection<String
+            }
+
+            if (!parserProperties.jointsToIgnore.isEmpty() && urdfModel.getJoints() != null)
+            {
+               urdfModel.getJoints().removeIf(joint -> parserProperties.jointsToIgnore.contains(joint.getName()));
+            }
+
+            if (!parserProperties.parseSensors && urdfModel.getGazebos() != null)
+            {
+               urdfModel.getGazebos().removeIf(gazebo -> gazebo.getSensor() != null);
+            }
+
+            if (parserProperties.handleImplicitJointDefinitions)
+               handleImplicitJointDefinitions(urdfModel);
+
+            // Merge the current URDFModel with the combined URDFModel
+            combinedURDFModel = mergeURDFModels(combinedURDFModel, urdfModel);
+         }
+      }
+      finally
+      {
+         for (InputStream inputStream : inputStreams)
          {
-            urdfModel = (URDFModel) um.unmarshal(inputStream);
+            try
+            {
+               inputStream.close();
+            }
+            catch (IOException e)
+            {
+               LogTools.error(e.getMessage());
+            }
          }
-         else
+      }
+
+      return combinedURDFModel;
+   }
+
+   /**
+    * Determine the parent model and child model, return a merged model that contains the parent model and all parts of the child model
+    * that aren't contained within the parent model. The name is taken from the parent model
+    * @param model1 URDF model with joint and links to be checked
+    * @param model2 URDF model with joints and links to be checked
+    * @return the merged model that contains all the joints, links, gazebos, and correct name from the models
+    */
+   private static URDFModel mergeURDFModels(URDFModel model1, URDFModel model2)
+   {
+      // Check the edge cases where a model might not have anything
+      if (model1.getLinks() == null && model1.getJoints() == null)
+      {
+         return model2;
+      }
+      else if (model2.getLinks() == null && model2.getJoints() == null)
+      {
+         return model1;
+      }
+
+      // Get all names of links in model1
+      List<String> model1LinkNames = new ArrayList<>();
+      List<String> model2LinkNames = new ArrayList<>();
+
+      if (model1.getLinks() != null)
+      {
+         for (URDFLink link : model1.getLinks())
          {
-            InputSource is = new InputSource(inputStream);
-            SAXParserFactory sax = SAXParserFactory.newInstance();
-            sax.setNamespaceAware(false);
-            XMLReader reader;
+            model1LinkNames.add(link.getName());
+         }
+      }
 
-            try
+      if (model2.getLinks() != null)
+      {
+         for (URDFLink link : model2.getLinks())
+         {
+            model2LinkNames.add(link.getName());
+         }
+      }
+
+      URDFModel parentModel = new URDFModel();
+      URDFModel childModel = new URDFModel();
```

Comment:
No need to instantiate new models, just needed more references, these could be initialized to null

Reply:
Done, yep true, should know that by now dang